src/EventListener/Login.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * Rcm Event Wrapper
+ *
+ * RCM Event Wrapper will listen for events and fire the correct CMS
+ * action for that event
+ *
+ * PHP version 5.3
+ *
+ * LICENSE: No License yet
+ *
+ * @category  Reliv
+ * @package   Rcm
+ * @author    Westin Shafer <wshafer@relivinc.com>
+ * @copyright 2012 Reliv International
+ * @license   License.txt New BSD License
+ * @version   GIT: <git_id>
+ * @link      http://github.com/reliv
+ */
+namespace RcmLogin\EventListener;
+
+use Zend\EventManager\Event;
+use Zend\Http\Response;
+
+/**
+ * Rcm Login Default Event Listener
+ *
+ * RCM Login Default Event Listener
+ *
+ * @category  Reliv
+ * @package   RcmLogin
+ * @author    Westin Shafer <wshafer@relivinc.com>
+ * @copyright 2012 Reliv International
+ * @license   License.txt New BSD License
+ * @version   Release: 1.0
+ * @link      http://github.com/reliv
+ */
+class Login
+{
+    /**
+     * LoginSuccess
+     *
+     * @param MvcEvent $event event
+     *
+     * @return Response
+     */
+    public function loginSuccess(Event $event)
+    {
+        /** @var \Zend\ServiceManager\ServiceManager $serviceManager */
+        $serviceManager = $event->getTarget()->getServiceLocator();
+
+        $config = $serviceManager->get('config');
+
+        /** @var $request \Zend\Http\Request */
+        $request = $serviceManager->get('request');
+
+        $redirect = $request->getUri()->toString();
+
+        if (!empty($config['rcmPlugin']['RcmLogin']['defaultSuccessRedirect'])) {
+            $redirect = $config['rcmPlugin']['RcmLogin']['defaultSuccessRedirect'];
+        }
+
+        $redirect = $request->getQuery('redirect', $redirect);
```

Comment:
Here is our current filtering on the redirect url 

    public function parseRedirectUrl($url)
    {
        return str_replace(['http://', 'https://', '//'], null, filter_var($url, FILTER_SANITIZE_URL));
    }